open_instruct/dataset_transformation.py

```diff
@@ -533,9 +690,11 @@ def get_tokenizer_tulu_v2_2(tc: "TokenizerConfig"):
             # OLMo newer models use this tokenizer
             if tokenizer.bos_token is None:
                 tokenizer.bos_token = tokenizer.eos_token
-                assert tc.add_bos, (
-                    "For OLMo with GPTNeoX, you must add bos token to the beginning of the input sequence."
-                )
+                if "olmo" not in tc.chat_template_name:
```

Comment:
Is this backward compatible? E.g. if someone tries to reproduce an olmo 2 run and messes something up, I don't think this will trigger?